src/Microsoft.OpenApi/Models/OpenApiDocument.cs

```diff
@@ -665,47 +665,52 @@ public bool AddComponent<T>(string id, T componentToRegister)
             Utils.CheckArgumentNull(componentToRegister);
             Utils.CheckArgumentNullOrEmpty(id);
             Components ??= new();
+
+            static Dictionary<string, TValue> AddToDictionary<TValue>(Dictionary<string, TValue>? dict, string key, TValue value)
+            {
+                dict ??= new Dictionary<string, TValue>();
```

Comment:
I'd argue that we should NOT initialize the dictionaries here as it:
- mixes concerns (add vs initialize) (readability)
- makes the method prototype more complex (readability)
- forces us to assign the dictionary value (on the component object) at every call (perf)